src/components/portfolio/Details.tsx

```diff
@@ -24,430 +30,104 @@ const useStyles = makeStyles(({ palette }: Theme) =>
-  [1359586800000, 37.83],
-  [1359673200000, 38.34],
-  [1359932400000, 38.1],
-  [1360018800000, 38.51],
-  [1360105200000, 38.4],
-  [1360191600000, 38.07],
-  [1360278000000, 39.12],
-  [1360537200000, 38.64],
-  [1360623600000, 38.89],
-  [1360710000000, 38.81],
-  [1360796400000, 38.61],
-  [1360882800000, 38.63],
-  [1361228400000, 38.99],
-  [1361314800000, 38.77],
-  [1361401200000, 38.34],
-  [1361487600000, 38.55],
-  [1361746800000, 38.11],
-  [1361833200000, 38.59],
-  [1361919600000, 39.6],
-];
-
 // props type declaration
 export type DetailsProps = {
+  token: string;
+  id: string;
   // function to return to the dashboard
   back: () => void;
 };
 
 // functional component that takes the name of the portfolio and a function to switch back to the dashboard
 // returns the entire details page
-const Details: React.FC<DetailsProps> = ({ back }) => {
+const Details: React.FC<DetailsProps> = ({ token, id, back }) => {
+  const [portfolioDetails, setPortfolioDetails] = React.useState<
+    API.PortfolioDetails | undefined
+  >();
+  const [error, setError] = React.useState<ErrorCode | undefined>(undefined);
+
+  const isMounted = React.useRef(true);
+
+  const fetch = async () => {
+    setError(undefined);
+    try {
+      const details = await API.details(token, id);
+      if (isMounted.current) {
+        setPortfolioDetails(details);
+      }
+    } catch (e) {
+      if (isMounted.current) {
+        setError(e.message);
+      }
+    }
+  };
+
+  React.useEffect(() => {
+    fetch();
+    return () => {
+      isMounted.current = false;
+    };
+    // deps must be empty because the function should only be called on mount.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   const classes = useStyles();
+  const { t } = useTranslation();
 
   return (
     <div>
-      <section className={classes.topBanner}>
+      <div className={classes.topBanner}>
         <DetailsHeader
           back={back}
-          name={mock.name}
-          positions={mock.positions}
-        />
-      </section>
-      <section className={classes.main}>
-        <DetailsMain
-          positionCount={mock.positionCount}
-          value={mock.value}
-          score={mock.score}
-          perf7d={mock.perf7d}
-          perf1y={mock.perf1y}
-          risk={mock.risk}
-          positions={mock.positions}
-          figures={mock.keyFigures}
-          nextDividend={mock.nextDividend}
-          dividendPayoutRatio={mock.dividendPayoutRatio}
-          performance={mockPerformance}
+          name={portfolioDetails?.overview.name}
+          positions={portfolioDetails?.positions}
+          editPositions={async (modifications) => {
+            await API.modify(token, id, modifications);
+            // reload portfolio details after successful edit
+            await fetch();
+          }}
         />
-      </section>
+      </div>
+      {!portfolioDetails && !error && <LinearProgress color="secondary" />}
```

Comment:
Wouldn't it be nice to change this to `portfolioDetails || error || <LinearProgress color="secondary" />}` ?

Reply:
`portfolioDetails` and `error` are not booleans, so these expressions are not the same. For example, if `portfolioDetails` is not `undefined`, `!portfolioDetails` evaluates to `false`, but `portfolioDetails` evaluates to a `PortfolioDetails` object. Because this object is truthy, React would try to render the object, causing an error.

See [this](https://reactjs.org/docs/conditional-rendering.html) and [this](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) for more information.